pkg/common/namespace_selection.go

```diff
@@ -103,3 +116,177 @@ func GetAllNamespaces(client kubernetes.Interface, labelSelector metav1.LabelSel
 
 	return namespacesNames, nil
 }
+
+type NamespaceSelectorReconciler struct {
+	Client     client.Client
+	selections map[string]NamespaceSelection
+	lock       sync.RWMutex
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *NamespaceSelectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.selections = make(map[string]NamespaceSelection)
+
+	neverEnqueue := predicate.NewPredicateFuncs(func(o client.Object) bool { return false })
+	mapToSingleton := func(_ client.Object) []reconcile.Request {
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{
+			Name: "NamespaceSelector",
+		}}}
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("NamespaceSelector").
+		For(
+			&policyv1.ConfigurationPolicy{},
+			builder.WithPredicates(neverEnqueue)).
```

Comment:
Will this be an issue in hosted mode since the configuration policy CRD would not be installed on the hosted cluster?